tests/sentry/integrations/github/test_issues.py

```diff
@@ -77,48 +87,86 @@ def test_get_allowed_assignees(self, mock_get_jwt):
             self._check_proxying()
 
     @responses.activate
-    @patch("sentry.integrations.github.client.get_jwt", return_value="jwt_token_1")
-    def test_get_repo_labels(self, mock_get_jwt):
+    @patch.object(sentry.integrations.github.client.GitHubClientMixin, "page_size", 1)
+    def test_get_labels_all_and_pagination(self):
+        """Fetch all labels and test the pagination logic."""
+        self.stub_access_token()
+
+        pp = 1
+        labels = [
+            {"name": "bug"},
+            {"name": "enhancement"},
+            {"name": "duplicate"},
+        ]
+
+        api_url = "https://api.github.com/repos/getsentry/sentry/labels"
+        first = f'<{api_url}?per_page={pp}&page=1>; rel="first"'
+        last = f'<{api_url}?per_page={pp}&page={len(labels)}>; rel="last"'
+
+        def gen_link(page: int, text: str) -> str:
+            return f'<{api_url}?per_page={pp}&page={page}>; rel="{text}"'
+
         responses.add(
-            responses.POST,
-            "https://api.github.com/app/installations/github_external_id/access_tokens",
-            json={"token": "token_1", "expires_at": "2018-10-11T22:14:10Z"},
+            responses.GET,
+            url=api_url,
+            match=[responses.matchers.query_param_matcher({"per_page": pp})],
+            json=labels[:1],
+            headers={"link": ", ".join([gen_link(2, "next"), last])},
         )
         responses.add(
             responses.GET,
-            "https://api.github.com/repos/getsentry/sentry/labels",
-            json=[{"name": "bug"}, {"name": "enhancement"}, {"name": "duplicate"}],
+            url=api_url,
+            match=[responses.matchers.query_param_matcher({"per_page": pp, "page": 2})],
+            json=labels[1:2],
+            headers={"link": ", ".join([gen_link(1, "prev"), gen_link(3, "next"), last, first])},
+        )
+        responses.add(
+            responses.GET,
+            url=api_url,
+            match=[responses.matchers.query_param_matcher({"per_page": pp, "page": 3})],
+            json=labels[-1:],
+            headers={"link": ", ".join([gen_link(2, "prev"), first])},
         )
 
-        repo = "getsentry/sentry"
-
-        # results should be sorted alphabetically
-        assert self.integration.get_repo_labels(repo) == (
+        response = self.integration.get_repo_labels(self.repo, fetch_max_pages=True)
+        assert response == (
             ("bug", "bug"),
             ("duplicate", "duplicate"),
             ("enhancement", "enhancement"),
         )
 
         if self.should_call_api_without_proxying():
-            assert len(responses.calls) == 2
+            assert len(responses.calls) == 6
```

Comment:
Normally this would always be 2, 1 req to get the access_token and another for the real req, but there are 3 real requests so I'm assuming 2 (calls per real req) * 3 (real reqs) = 6 calls total